OpenRA.Game/Traits/Player/FrozenActorLayer.cs

```diff
@@ -179,7 +181,7 @@ public void Tick(Actor self)
 		public virtual IEnumerable<IRenderable> Render(Actor self, WorldRenderer wr)
 		{
 			return world.ScreenMap.FrozenActorsInBox(owner, wr.Viewport.TopLeft, wr.Viewport.BottomRight)
-				.Where(f => f.Visible)
+				.Where(f => f.Visible && !owner.Shroud.IsUnderDisruptionField(f.OccCells))
```

Comment:
Like I mentioned above, your disruption checks should be built into `f.Visible` instead of adding IsUnderDisruptionField everywhere.